src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/SimpleFileServerImpl.java

```diff
@@ -169,10 +170,19 @@ void printStartMessage(Path root, HttpServer server)
             if (!addrSpecified) {
                 writer.println(ResourceBundleHelper.getMessage("loopback.info"));
             }
+            String url;
+            if (inetAddr instanceof Inet6Address && addr.contains(":") && !addr.startsWith("[")) {
+                // RFC-2732, section 2:
+                // To use a literal IPv6 address in a URL, the literal
+                // address should be enclosed in "[" and "]" characters.
+                url = "http://[" + addr + "]:" + port + "/";
+            } else {
+                url = "http://" + addr + ":" + port + "/";
+            }
             if (isAnyLocal) {
-                writer.println(ResourceBundleHelper.getMessage("msg.start.anylocal", root, addr, port));
+                writer.println(ResourceBundleHelper.getMessage("msg.start.anylocal", root, port, url));
             } else {
-                writer.println(ResourceBundleHelper.getMessage("msg.start.other", root, addr, port));
+                writer.println(ResourceBundleHelper.getMessage("msg.start.other", root, addr, port, url));
```

Comment:
I would suggest simply adding '[' and ']' around the address in that case - `addr = '[' + addr + ']';` which would allow to leave the resource bundle unchanged.

Reply:
In addition to using that param value in the URL, that resource bundle message also uses it in another message:

```
Serving /tmp/foo/ and subdirectories on 0:0:0:0:0:0:0:1 port 8000
```
If we change the `addr` being passed here to be `[` + addr + `]`, then this above message would be:

```
Serving /tmp/foo/ and subdirectories on [0:0:0:0:0:0:0:1] port 8000
```
Is that fine?

Reply:
Oh - I see... Hmmm - good question. I'd say it should be OK, since it's only a log message.
My concern here is potentially backporting this change - where we would need to make sure that all resource bundles in all possibles languages that are supported are correctly updated.

Reply:
> My concern here is potentially backporting this change - where we would need to make sure that all resource bundles in all possible languages that are supported are correctly updated.

I hadn't considered that. Given that, I think it's fine to just enclose the other line too between `[` and `]`. Like you note, that's just a log message anyway. I'll update the PR shortly.

Reply:
Done. The test continues to pass.